src/libs/web-preset.ts

```diff
@@ -51,10 +53,28 @@ const registerOnReconnect = (cb: () => void) => {
   }
 }
 
+const IS_SERVER =
+  typeof window === 'undefined' ||
+  // @ts-ignore
+  !!(typeof Deno !== 'undefined' && Deno && Deno.version && Deno.version.deno)
+
+// polyfill for requestAnimationFrame
+const rAF = IS_SERVER
+  ? null
+  : window['requestAnimationFrame'] || (f => setTimeout(f, 1))
+
+// React currently throws a warning when using useLayoutEffect on the server.
+// To get around it, we can conditionally useEffect on the server (no-op) and
+// useLayoutEffect in the browser.
+const useIsomorphicLayoutEffect = IS_SERVER ? useEffect : useLayoutEffect
+
 export default {
   isOnline,
   isDocumentVisible,
   fetcher,
   registerOnFocus,
-  registerOnReconnect
+  registerOnReconnect,
+  rAF,
+  useIsomorphicLayoutEffect,
+  IS_SERVER
```

Comment:
I think these belong to web-preset, but shouldn't be included in `defaultConfig` because they are not "config", and we don't want to expose them to the context.

Reply:
@shuding Thank you for your feedback! That makes sense 👍  What about exposing them as an external file like `env.ts`? Should I copy and have them as duplicated?

Reply:
> What about exposing them as an external file like env.ts?

Yeah that’s a good idea 👍

Reply:
Thank you! I've fixed it.